### PR TITLE
chore: expose adb devices and actions

### PR DESCRIPTION
**Pull request by pavelfeldman**
https://github.com/microsoft/playwright/issues/1122

### CODE REVIEW

File: packages/build_package.js [hunk 1]
```diff
@@ -64,6 +64,12 @@ const PACKAGES = {
     browsers: [],
     files: [...PLAYWRIGHT_CORE_FILES, ...FFMPEG_FILES, 'electron-types.d.ts'],
   },
+  'playwright-android': {
+    version: '0.0.2', // Manually manage playwright-android version.
+    description: 'A high-level API to automate Chrome for Android',
+    browsers: [],
+    files: [...PLAYWRIGHT_CORE_FILES, ...FFMPEG_FILES, 'android-types.d.ts', 'bin/android-driver.apk', 'bin/android-driver-target.apk'],
```
Comment (dgozman) [inline packages/build_package.js:71]: Please take a look at npmignore file. I think we are either shipping apks to all packages, or ignoring them in playwright-android.

File: src/client/types.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ export type Headers = { [key: string]: string };
 export type Env = { [key: string]: string | number | boolean | undefined };
 export type URLMatch = string | RegExp | ((url: URL) => boolean);
 
+export type TimeoutOptions = Function | { timeout?: number };
```
Comment (dgozman) [inline src/client/types.ts:33]: Why `Function` in timeout options?

File: src/server/android/android.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    debug('pw:android')('Starting the new driver');
+    this.shell(`am instrument -w com.microsoft.playwright.androiddriver.test/androidx.test.runner.AndroidJUnitRunner`);
+
+    debug('pw:android')('Polling the socket');
+    let socket;
+    while (!socket) {
+      try {
+        socket = await this._backend.open(`localabstract:playwright_android_driver_socket`);
+      } catch (e)  {
+        await new Promise(f => setTimeout(f, 100));
```
Comment (dgozman) [inline src/server/android/android.ts:121]: I guess we'll need a timeout for this operation.

File: src/server/android/android.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    const browser = await CRBrowser.connect(androidBrowser, browserOptions);
+    const controller = new ProgressController();
+    await controller.run(async progress => {
+      await browser._defaultContext!._loadDefaultContext(progress);
+    });
+    return browser._defaultContext!;
+  }
+}
+
+class AndroidBrowser extends EventEmitter {
```
Comment (dgozman) [inline src/server/android/android.ts:200]: Shall we mark this as `implements Transport`? IIUC, this serves as a transport for protocol connection?

File: src/server/android/android.ts [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+      protocolLogger: helper.debugProtocolLogger(),
+      browserLogsCollector: new RecentLogsCollector()
+    };
+    validateBrowserContextOptions(options, browserOptions);
+
+    const browser = await CRBrowser.connect(androidBrowser, browserOptions);
+    const controller = new ProgressController();
+    await controller.run(async progress => {
+      await browser._defaultContext!._loadDefaultContext(progress);
+    });
```
Comment (dgozman) [inline src/server/android/android.ts:195]: For proper timeout managing, you have to wrap everything in this method into `controller.run` and pass `TimeoutSettings.timeout(options)` to it.

File: utils/check_deps.js [hunk 6]
```diff
@@ -111,10 +113,11 @@ DEPS['src/server/common/'] = [];
 DEPS['src/server/injected/'] = ['src/server/common/'];
 
 // Electron and Clank use chromium internally.
+DEPS['src/server/android/'] = [...DEPS['src/server/'], 'src/server/chromium/', 'src/protocol/transport.ts'];
 DEPS['src/server/electron/'] = [...DEPS['src/server/'], 'src/server/chromium/'];
 DEPS['src/server/clank/'] = [...DEPS['src/server/'], 'src/server/chromium/'];
```
Comment (dgozman) [inline utils/check_deps.js:118]: Drop this line now?

**Review by dgozman — APPROVED** [review]

File: src/server/android/android.ts (hunk 5 above)
Comment (pavelfeldman) [reply src/server/android/android.ts:195]: Doing it in a follow up...

File: utils/check_deps.js (hunk 6 above)
Comment (pavelfeldman) [reply utils/check_deps.js:118]: Nothing works w/o it!

**Pull request merged**